### PR TITLE
ShellCheck: Add DistributionRequirement

### PR DESCRIPTION
**Pull request by naveentata**
We can find version"packages.ubuntu.com/xenial/shellcheck"

Closes:https://github.com/coala/coala-bears/issues/1359

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/shell/ShellCheckBear.py [hunk 1]
```diff
@@ -14,6 +16,8 @@ class ShellCheckBear:
     """
 
     LANGUAGES = {'sh', 'bash', 'ksh', 'dash'}
+    REQUIREMENTS = {
+          DistributionRequirement(apt_get='shellcheck', version='0.3.7')}
```
Comment (vijeth-aradhya) [inline bears/shell/ShellCheckBear.py]: I think it looks better this way :)
```
REQUIREMENTS = {DistributionRequirement(apt_get='shellcheck',
                                        version='0.3.7')}
```

**Comment by Techievena** [comment]
ack 74bc5f2

**Comment by Techievena** [comment]
unack 74bc5f2

**Comment by rohanraju97** [comment]
LGTM!
ack 21bf283061e0b490cf123157a9d5506188c088b2

**Comment by rohanraju97** [comment]
ack 21bf283061e0b490cf123157a9d5506188c088b2

**Comment by NiklasMM** [comment]
ack 21bf283

**Comment by jayvdb** [comment]
unack 21bf283

**Comment by jayvdb** [comment]
Misspelling in the commit message

**Comment by jayvdb** [comment]
also please rebase your commit

**Comment by naveentata** [comment]
@jayvdb `Shellcheck 0.3.7 is chosen as xenial ships that version` Is this commit body good?

**Comment by supergr35** [comment]
In my opinion, something more like:
```
Choose Shellcheck 0.3.7 as xenial ships this version.
```
This commit body is now in the imperative mood and describes the change made.

**Comment by supergr35** [comment]
Rebase and tests should pass

**Comment by yash-nisar** [comment]
Can you rebase so that circle ci tests pass ? @naveentata

**Comment by adhikasp** [comment]
The original cabal require shellcheck version `0.4.1` while this apt use version `0.3.7`. Does it have potential breakage?

**Comment by Mixih** [comment]
@naveentata why don't we use cabalrequirement for this?

File: .ci/deps.sh [hunk 2]
```diff
@@ -101,6 +101,7 @@ go get -u golang.org/x/tools/cmd/goimports
 go get -u sourcegraph.com/sqs/goreturns
 go get -u github.com/jayvdb/gotype
 go get -u github.com/kisielk/errcheck
+go get -u github.com/shurcooL/markdownfmt
```
Comment (supergr35) [inline .ci/deps.sh]: A fix for tests? Or a different reason.

Comment (Mixih) [inline .ci/deps.sh]: unrelated change, should go into a separate commit. If it fixes the test, split it out to a commit before the feature commit

**Comment by naveentata** [comment]
@Mixih As per this comment https://github.com/coala/coala-bears/pull/1367#issuecomment-277652300 
by @jayvdb  we should only use Distribution Requirement.

File: bears/shell/ShellCheckBear.py [hunk 3]
```diff
@@ -14,6 +16,8 @@ class ShellCheckBear:
     """
 
     LANGUAGES = {'sh', 'bash', 'ksh', 'dash'}
+    REQUIREMENTS = {DistributionRequirement(apt_get='shellcheck',
+                                            version='0.3.7')}
```
Comment (jayvdb) [inline bears/shell/ShellCheckBear.py]: `DistributionRequirement` doesnt support a version argument.

File: bears/shell/ShellCheckBear.py [hunk 4]
```diff
@@ -14,6 +16,7 @@ class ShellCheckBear:
     """
 
     LANGUAGES = {'sh', 'bash', 'ksh', 'dash'}
+    REQUIREMENTS = {DistributionRequirement(apt_get='shellcheck')}
```
Comment (jayvdb) [inline bears/shell/ShellCheckBear.py]: You can now use [AnyOneOfRequirements](https://gitlab.com/coala/package_manager/blob/master/dependency_management/requirements/AnyOneOfRequirements.py) to wrap these two requirements in an OR statement.

**Comment by jayvdb** [comment]
ack db09c29

**Comment by jayvdb** [comment]
ack db09c29

**Comment by jayvdb** [comment]
@rultor merge
